### PR TITLE
is_hot refactor and heating tweaks

### PR DESCRIPTION
**Pull request by TDSSS**
## What Does This PR Do
This PR does two things. First, it turns is_hot from an unsorted helper proc to an object level proc to remove that horrible istype construct. Secondly, adjusts how things like heaters and welding tools heat up reagents in a container, namely the more reagents you are heating up the slower you heat them. I also tweaked the temperatures for some of those heating sources, correcting lighters, matches and cigarettes down and welding tools up.

## Why It's Good For The Game
Removes weird code, makes lighters less OP at heating chemicals, adds muh realism.

## Changelog
:cl:
tweak: refactor of is_hot
tweak: heating up reagents with hot items takes longer if more reagents are being heated
/:cl:

### CODE REVIEW

File: code/_onclick/item_attack.dm [hunk 1]
```diff
@@ -12,9 +12,9 @@
 	return
 
 /obj/item/proc/pre_attackby(atom/A, mob/living/user, params) //do stuff before attackby!
-	if(is_hot(src) && A.reagents && !ismob(A))
+	if(is_hot() && A.reagents && !ismob(A))
 		to_chat(user, "<span class='notice'>You heat [A] with [src].</span>")
-		A.reagents.temperature_reagents(is_hot(src))
+		A.reagents.temperature_reagents(src.is_hot(), A.reagents.total_volume)	//higher volume means you heat it less
```
Comment (SteelSlayer) [inline code/_onclick/item_attack.dm]: Implied src on `src.is_hot()`

Comment (TDSSS) [reply code/_onclick/item_attack.dm]: right, duh. Thanks

File: code/_onclick/item_attack.dm [hunk 2]
```diff
@@ -12,9 +12,9 @@
 	return
 
 /obj/item/proc/pre_attackby(atom/A, mob/living/user, params) //do stuff before attackby!
-	if(is_hot(src) && A.reagents && !ismob(A))
+	if(is_hot() && A.reagents && !ismob(A))
 		to_chat(user, "<span class='notice'>You heat [A] with [src].</span>")
-		A.reagents.temperature_reagents(is_hot(src))
+		A.reagents.temperature_reagents(is_hot(), A.reagents.total_volume)	//higher volume means you heat it less
```
Comment (Fox-McCloud) [inline code/_onclick/item_attack.dm]: Please drop this change; I'd like to maintain quirks/oddities/gameplay with chems and heat with Goon.

**Comment by Fox-McCloud** [comment]
I'd recommend this be `get_temperature()` instead of `is_hot` to maintain parity with TG; also more descriptive.

**Comment by TDSSS** [comment]
I can do the renaming. For the second change, I am not sure why keeping things the same as goon just because is a good thing, tho. Not that it matters a huge deal either way.

**Comment by Fox-McCloud** [comment]
Keeping things at parity with Goon means allowing for a lot of interesting quirks and emergent gameplay--things I'm interested in keeping intact because it creates for a lot more interesting scenarios.

Either case, conflicts.

**Comment by TDSSS** [comment]
Yeah, tools refactor created a bunch of conflicts, will have to dig through and fix them, then do some testing when I have the time.

**Comment by TDSSS** [comment]
Might just start this over from scratch.

**Pull request closed without merging**